src/rdpc/query/fetchVariantCallingAnalyses.ts

```diff
@@ -0,0 +1,98 @@
+import { EgoAccessToken, EgoJwtManager } from "auth";
+import logger from "logger";
+import fetch from "node-fetch";
+import promiseRetry from "promise-retry";
+import { Analysis, AnalysisType } from "../types";
+import { QueryVariable, retryConfig } from "./types";
+
+const query = `
+query ($analysisFilter: AnalysisFilter, $analysisPage: Page){
+    analyses (
+      filter: $analysisFilter,
+      page: $analysisPage,
+      sorts:{fieldName:analysisId, order: asc}
+
+    ) {
+      content {
+        analysisId
+        analysisType
+        firstPublishedAt
+
+        workflow{
+          workflowName
+        }
+
+        donors {
+          donorId
+        }
+      }
+    }
+}
+`;
+
+const fetchVariantCallingAnalyses = async ({
+  studyId,
+  rdpcUrl,
+  from,
+  size,
+  egoJwtManager,
+  donorId,
+}: {
+  studyId: string;
+  rdpcUrl: string;
+  from: number;
+  size: number;
+  egoJwtManager: EgoJwtManager;
+  donorId?: string;
+}): Promise<Analysis[]> => {
+  const jwt = (await egoJwtManager.getLatestJwt()) as EgoAccessToken;
+  const accessToken = jwt.access_token;
+  return await promiseRetry<Analysis[]>(async (retry) => {
+    try {
+      const response = await fetch(rdpcUrl, {
+        method: "POST",
+        body: JSON.stringify({
+          query,
+          variables: {
+            analysisFilter: {
+              analysisState: "PUBLISHED",
+              analysisType: AnalysisType.VARIANT_CALLING,
+              studyId,
+              donorId,
+            },
+            analysisPage: {
+              from,
+              size,
+            },
+          } as QueryVariable,
+        }),
+        headers: {
+          "Content-type": "application/json",
+          authorization: `Bearer ${accessToken}`,
+        },
+      });
+      const jsonResponse = await response.json();
+      const hasError = jsonResponse.errors?.length > 0;
+      if (hasError) {
+        const error = JSON.stringify(jsonResponse.errors);
+        logger.error(
+          `received error from rdpc... page: from => ${from} size => ${size}. Error: ${error}`
+        );
+      }
+      const result = jsonResponse.data.analyses.content as Analysis[];
```

Comment:
Should we continue processing even if we found an error? All that we did was log the error then continue with the results anyways...

Reply:
The case I am concerned about is when `.fetch` does not throw an error, so we do not go to the catch block. Instead, we see `jsonResponse.errors?.length >0` so we log the errors we see (line 75-81), but then we continue processing the result on line 82 anyways. Is that correct to do?

This scenario will not result in a retry because we are not throwing an error, we are instead returning the result.